src/OrchardCore.Modules/OrchardCore.Users/Controllers/ChangeEmailController.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Logging;
+using OrchardCore.DisplayManagement;
+using OrchardCore.Email;
+using OrchardCore.Entities;
+using OrchardCore.Modules;
+using OrchardCore.Settings;
+using OrchardCore.Users.Models;
+using OrchardCore.Users.Services;
+using OrchardCore.Users.ViewModels;
+
+namespace OrchardCore.Users.Controllers
+{
+    [Feature("OrchardCore.Users.ChangeEmail")]
+    public class ChangeEmailController : Controller
+    {
+        private readonly IUserService _userService;
+        private readonly UserManager<IUser> _userManager;
+        private readonly ISiteService _siteService;
+        private readonly ILogger<ChangeEmailController> _logger;
+        private readonly IStringLocalizer<ChangeEmailController> S;
+
+        public ChangeEmailController(
+            IUserService userService,
+            UserManager<IUser> userManager,
+            ISiteService siteService,
+            ISmtpService smtpService,
+            IDisplayHelper displayHelper,
+            IStringLocalizer<ChangeEmailController> stringLocalizer,
+            ILogger<ChangeEmailController> logger)
+        {
+            _userService = userService;
+            _userManager = userManager;
+            _siteService = siteService;
+
+            S = stringLocalizer;
+            _logger = logger;
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> Index()
+        {
+            if (!(await _siteService.GetSiteSettingsAsync()).As<ChangeEmailSettings>().AllowChangeEmail)
+            {
+                return NotFound();
+            }
+
+            var user = await _userService.GetAuthenticatedUserAsync(User);
+
+            return View(new ChangeEmailViewModel() { Email = ((User)user).Email });
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Index(ChangeEmailViewModel model)
+        {
+            if (!(await _siteService.GetSiteSettingsAsync()).As<ChangeEmailSettings>().AllowChangeEmail)
+            {
+                return NotFound();
+            }
+
+            if (ModelState.IsValid)
+            {
+                var user = await _userService.GetAuthenticatedUserAsync(User);
+                var userWithEmail = await _userManager.FindByEmailAsync(model.Email);
+
+                if (((User)user).Email == model.Email)
+                {
+                    ModelState.AddModelError("Email", S["This email is already your current one."]);
+                }
+                else if (userWithEmail != null && user.UserName != userWithEmail.UserName)
+                {
+                    ModelState.AddModelError("Email", S["A user with the same email already exists."]);
```

Comment:
Return the same error as when a new account with an existing email is created.

Reply:
Do you mean 'Add the same email check in the RegistrationController'?